migrate_legacy_captures.js

```diff
@@ -0,0 +1,218 @@
+/**
+ *  This is a migration script that is to get all the records from legacy `public.tree` and populate
+ *  table `treetracker.capture` as long as the legacy records in `public.trees` are approved,
+ *  active, has a valid image url and if legacy data has not made it to capture table already.
+ *  In addition, if the concept of planter referenced in legacy trees table is not found in the
+ *  `treetracker.grower_account`, it is created first using the data from `public.planter` before proceeding
+ *  with populating the `treetracker.captures` table.
+ *
+ *  @created Jan 28 2022
+ *
+ * */
+
+require("dotenv").config();
+const generate_uuid_v4 = require("uuid").v4;
+
+const { knex } = require("./database/knex");
+const Writable = require("stream").Writable;
+const ws = Writable({ objectMode: true });
+const ProgressBar = require("progress");
+
+async function ensureGrowerAccountExists(legacyTree, transaction) {
+  let planter = null;
+  if (!legacyTree.planter_identifier && !legacyTree.planter_id) {
+    throw Error("No planter record found in legacy tree table");
+  }
+  if (legacyTree.planter_id) {
+    planter = await transaction
+      .select()
+      .table("public.planter")
+      .where("id", legacyTree.planter_id)
+      .first();
+  } else if (
+    legacyTree.planter_identifier &&
+    legacyTree.planter_identifier.trim()
+  ) {
+    planter = await transaction
+      .select()
+      .table("public.planter")
+      .where("email", legacyTree.planter_identifier)
+      .orWhere("phone", legacyTree.planter_identifier)
+      .first();
+  }
+  if (!planter) {
+    throw Error(
+      `No planter record found for tree entry with id ${legacyTree.id}`
+    );
+  }
+  let growerAccount = await transaction("treetracker.grower_account")
+    .where("email", planter.email)
+    .orWhere("phone", planter.phone)
+    .first();
+  if (!growerAccount) {
+    console.log(
+      `GrowerAccount not found for planter_identifier ${legacyTree.planter_identifier}`
+    );
+    console.log(
+      `Populating grower account for ${legacyTree.planter_identifier}`
+    );
+    // status, first_registration_at aren't available in the source `public.planter` table

```

Comment:
the default for status is 'active', which is also the DB default. The column data type is an enum of 'active' and 'deleted'